libpod/container.go

```diff
@@ -209,6 +209,9 @@ type ContainerState struct {
 	// healthcheck. The container will be restarted if this exceed a set
 	// number in the startup HC config.
 	StartupHCFailureCount int `json:"startupHCFailureCount,omitempty"`
+	// HCUnitName records the name of the healthcheck unit.
+	// Ref: RH Jira RHEL-26105
```

Comment:
tbh not sure how the issue number should help a reader here on the field definition. it would make more sense where we create a random unit name IMO